code-challenges/quick-sort/quick-sort.test.js

```diff
@@ -0,0 +1,98 @@
+
+function quickSort(array, leftLimit = 0, rightLimit = array.length - 1) {
+
+  // only do any of this if the array length is greater than 1 (this is the base case).
+  if (leftLimit >= rightLimit) return array;
+
+  //first partition the array, returning the pivotIndex (the index of the pivot after the rough sort)
+  const pivotIndex = partition(array, leftLimit, rightLimit);
+
+  //then make a recursive call to the left
+  quickSort(array, leftLimit, pivotIndex - 1)
+
+  //then make a recursive call to the right
+  quickSort(array, pivotIndex, rightLimit)
```

Comment:
Do we need to include pivotIndex here? The entry at pivotIndex is already in the correct place.

Reply:
great catch !